modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -2617,6 +2617,8 @@ void ONNXImporter::parseConcat(LayerParams& layerParams, const opencv_onnx::Node
 
         // Concat-1 has default value for axis is 1: https://github.com/onnx/onnx/blob/master/docs/Changelog.md#Concat-1
         int axis = layerParams.get<int>("axis", 1);
+        // Concat with negative axis is not supported for now: https://github.com/opencv/opencv/issues/24861
+        CV_Assert(axis >= 0);
```

Comment:
Can axis be normalized instead? At least to cover ONNX models for now
```
axis = normalize_axis(axis, inputShape.size());
```

Reply:
Normalizing axes in importer may cause problem as input shape may change. If we narrow down the talk to ConvNets, it should be safe enough. If we pursue a safer solution, axes normalization should be done in `finalize()`.